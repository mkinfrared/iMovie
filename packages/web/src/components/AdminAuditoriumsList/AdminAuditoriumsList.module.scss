@import "src/styles/common";

.AdminAuditoriumsList {
  @extend .gridList;
}
