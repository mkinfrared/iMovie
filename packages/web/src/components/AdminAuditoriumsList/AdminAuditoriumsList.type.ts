import { ReduxConnectedComponent } from "react-app-env";
import { Auditorium } from "store/reducers/cinemas/types";

export interface AdminAuditoriumsListProps extends ReduxConnectedComponent {
  cinemaId: number;
  auditoriums: Auditorium[];
}
