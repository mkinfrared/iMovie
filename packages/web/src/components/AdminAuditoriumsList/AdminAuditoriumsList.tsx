import React, { useEffect } from "react";

import AuditoriumCard from "components/AuditoriumCard";
import { fetchCinemaAuditoriums } from "store/reducers/cinemas/actions";

import css from "./AdminAuditoriumsList.module.scss";
import { AdminAuditoriumsListProps } from "./AdminAuditoriumsList.type";

const AdminAuditoriumsList = ({
  cinemaId,
  auditoriums,
  dispatch
}: AdminAuditoriumsListProps) => {
  useEffect(() => {
    dispatch(fetchCinemaAuditoriums(cinemaId));
  }, []);

  const list = auditoriums.map(({ id, name, seats }) => (
    <AuditoriumCard
      key={id}
      name={name}
      seats={seats.length}
      cinemaId={cinemaId}
      auditoriumId={id}
    />
  ));

  return <div className={css.AdminAuditoriumsList}>{list}</div>;
};

export default React.memo(AdminAuditoriumsList);
