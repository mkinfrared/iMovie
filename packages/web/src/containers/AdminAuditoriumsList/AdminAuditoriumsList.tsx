import React from "react";
import { useDispatch, useSelector } from "react-redux";

import AdminAuditoriumsListFC from "components/AdminAuditoriumsList";
import { AdminAuditoriumsListProps } from "components/AdminAuditoriumsList/AdminAuditoriumsList.type";
import { selectCinemaAuditoriums } from "store/reducers/cinemas/selectors";

const AdminAuditoriumsList = ({
  cinemaId
}: Pick<AdminAuditoriumsListProps, "cinemaId">) => {
  const dispatch = useDispatch();
  const auditoriums = useSelector(selectCinemaAuditoriums(cinemaId)) ?? [];

  return (
    <AdminAuditoriumsListFC
      cinemaId={cinemaId}
      auditoriums={auditoriums}
      dispatch={dispatch}
    />
  );
};

export default React.memo(AdminAuditoriumsList);
