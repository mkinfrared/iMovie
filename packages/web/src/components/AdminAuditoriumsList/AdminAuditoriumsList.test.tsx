import { render } from "@testing-library/react";
import React, { ReactElement } from "react";
import { MemoryRouter } from "react-router-dom";

import AdminAuditoriumsList from "./AdminAuditoriumsList";

describe("<AdminAuditoriumsList />", () => {
  const dispatch = jest.fn();
  const cinemaId = 42;

  const auditorium = {
    id: 1,
    name: "Mercury",
    cinemaId: 1,
    createdAt: "2020-05-17T08:16:44.603Z",
    updatedAt: "2020-05-17T08:16:44.603Z",
    seats: [
      {
        id: 1,
        row: "A",
        number: 1,
        auditoriumId: 1
      }
    ]
  };

  const auditoriums = [auditorium];

  let Component: ReactElement;

  beforeEach(() => {
    Component = (
      <MemoryRouter>
        <AdminAuditoriumsList
          dispatch={dispatch}
          cinemaId={cinemaId}
          auditoriums={auditoriums}
        />
      </MemoryRouter>
    );
  });

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match a snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
