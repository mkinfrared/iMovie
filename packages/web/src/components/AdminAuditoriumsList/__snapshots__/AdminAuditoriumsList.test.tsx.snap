// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminAuditoriumsList /> should match a snapshot 1`] = `
<div>
  <div
    class="AdminAuditoriumsList"
  >
    <div
      class="MuiPaper-root MuiCard-root AuditoriumCard MuiPaper-outlined MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
        >
          Mercury
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          Details
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom"
        >
          Seats:
           
          <span
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            1
          </span>
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <a
          href="/admin/cinema/42/1/edit"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              More
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
